yjit/src/codegen.rs

```diff
@@ -993,6 +993,47 @@ fn gen_putobject_int2fix(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlo
     KeepCompiling
 }
 
+// set Nth stack entry to stack top
+fn gen_setn(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock, ocb: &mut CodeBlock) -> CodegenStatus
+{
+    let nval:VALUE = jit_get_arg(jit, 0);
+    let VALUE(n) = nval;
+
+    let top_val:X86Opnd = ctx.stack_pop(0);
+    let dst_opnd:X86Opnd = ctx.stack_opnd(n.try_into().unwrap());
+    mov(cb, REG0, top_val);
+    mov(cb, dst_opnd, REG0);
+
+    let mapping = ctx.get_opnd_mapping(InsnOpnd::StackOpnd(0));
```

Comment:
Will merge but let's put `use InsnOpnd::*` at the top of the file.

I also think the code would be more readable if n was converted into the right input type for stack_opnd earlier.

Reply:
With respect to the type conversion, I think you had the right approach. I apologize.